SecretManagement.KeePass.Extension/Tests/Test-SecretVault.Tests.ps1

```diff
@@ -102,29 +142,509 @@ InModuleScope -ModuleName 'SecretManagement.KeePass.Extension' {
                 } catch [system.Exception] { }
             }
-                It "should not have a variable 'Vault_$($VaultName)'" {
+            It "should not have a variable 'Vault_$($VaultName)'" {
                 { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
             }
             It "should request a credential on the first pass" {
                 Test-SecretVault -VaultName $VaultName
-                Assert-MockCalled -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+                Should -Invoke -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+            }
+            It 'Should not request a credential on the second pass' {
+                Test-SecretVault -VaultName $VaultName
+                Should -Invoke -CommandName 'Get-Credential' -Exactly 1 -Scope Context
+            }
+            It "should have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Not -Throw
+            }
+        }
+        Context "Validating with Path and incorrect UseMasterPassword" {
+            BeforeAll {
+                $MasterKey = 'You can not enter with this masterkey'
+                $VaultMasterKey = [PSCredential]::new('vaultkey', (ConvertTo-SecureString -AsPlainText -Force $MasterKey))
+
+                $VaultName = "KeepassPesterTest_$([guid]::NewGuid())"
+                $KeePassDatabaseSuffix = 'PathOnly'
+                $KeePassDatabaseFileName = "$($BaseKeepassDatabaseName)$($KeePassDatabaseSuffix).kdbx"
+                $VaultPath = Join-Path -Path $TestDrive -ChildPath $KeePassDatabaseFileName
+                Copy-Item -Path "$($PSScriptRoot)/$($KeePassDatabaseFileName)" -Destination $VaultPath
+
+                $RegisterSecretVaultPathOnlyParams = @{
+                    Name            = $VaultName
+                    ModuleName      = $ModulePath
+                    PassThru        = $true
+                    VaultParameters = @{
+                        Path = $VaultPath
+                    }
+                }
+                Microsoft.PowerShell.SecretManagement\Register-SecretVault @RegisterSecretVaultPathOnlyParams | Out-Null
+
+                Mock -Verifiable -CommandName 'Get-Credential' -MockWith { $VaultMasterKey }
+            }
+            AfterAll {
+                try {
+                    Microsoft.PowerShell.SecretManagement\Get-SecretVault -Name $VaultName -ErrorAction SilentlyContinue | Microsoft.PowerShell.SecretManagement\Unregister-SecretVault -ErrorAction SilentlyContinue
+                } catch [system.Exception] { }
+            }
+            It "should not have a variable 'Vault_$($VaultName)'" {
+                { (Get-Variable -Name "Vault_$VaultName" -Scope Script).Value } | Should -Throw
+            }
+            It "should throw on the first pass" {
+                { Test-SecretVault -VaultName $VaultName } | should -Throw
```

Comment:
-Throw should be relatively explicit to the expected exception, in this case: | Should -Throw '*The composite key is invalid!*'

Reply:
> @trir262 Looks good, only ask is to make the throws more specific to the exception you are expecting, so if you get a different exception the test still fails instead of it being "swallowed"

It is not an issue for the empty values, but "the composite key" message is not so easy.

Reply:
@trir262 can you elaborate why it is "not so easy"? Not sure what you mean

Reply:
Turned out it wasn't so difficult after all. The error message that is returned when the keepass key is not correct has both newlines and some quotes in the text. It took me awhile before finding out what is 'the exact' message and make it work on both windows & non-windows.
They are now also tested (I renamed my tests for clarity as well)

Reply:
All done for this now... new branch for tests on get-Secret is on the way ;)

Reply:
simplified the composite key error message